aten/src/ATen/native/cpu/UpSampleMoreKernel.cpp

```diff
@@ -14,6 +14,70 @@ namespace {
 
 using scale_t = std::vector<c10::optional<double>>;
 
+template <typename scalar_in, typename scalar_out>
+void inline nearest_channels_last_acc(scalar_in* gin, scalar_out* gout, int64_t size) {
+  using Vec = vec::Vectorized<scalar_in>;
```

Comment:
`Vec::size()` of `scalar_in` and `scalar_out` might not match. Have you considered this case?

Reply:
scalar_in and scalar_out are always same when the data type is not BFloat16.

Reply:
Do you mean `scalar_in` and `scalar_out` are always the same?

Reply:
I mean there are two cases to use `nearest_channels_last_acc` : 1. scalar_in and scalar_out are same; 2. scalar_in=float and scalar_out =BFloat16.
 When opmath_t is not scalar_t i.e., scalar_t=BFloat16 and opmath_t=float `template <>
void inline nearest_channels_last_acc(float* gin, BFloat16* gout, int64_t size)` is called. 
When opmath_t == scalar_t i.e., scalar_t=float or double `template <typename scalar_in, typename scalar_out>
void inline nearest_channels_last_acc(scalar_in* gin, scalar_out* gout, int64_t size)` is called.

Reply:
Please check they are equal here.

Reply:
Added check.